<div style="border-radius: 15px; border: 3px solid indigo; padding: 15px;">
<b> Комментарий ревьюера</b>
    
Привет! Прошу прощения за задержку ревью 🙏 Понимаю, что это принесло много неудобств, но скоро все будет в норме. 
    
    
Меня зовут Светлана Носова, я буду проверять твой проект. Если не против, давай общаться на «ты». Если тебе такой стиль общения не нравится, напиши об этом, будем на «вы».

Сразу хочу обратить внимание на цветовую разметку:

<div class="alert alert-success" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">

Зеленым с пометкой ✔️ отмечены отличные решения и идеи, которые можно и нужно использовать в будущем.
</div>
    
<div class="alert alert-warning" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">

Желтым с пометкой ⚠️ отмечено то, что можно или нужно оптимизировать, в зависимости от контекста.</div>    
    
<div class="alert alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">

Красным с пометкой ❌ обозначены пропущенные пункты задания и ошибки, которые обязательно надо будет исправить, чтобы результаты были верные.</div> 
    
 **Ответы на мои комментарии или доработки по моим рекомендациям попрошу выделить каким-нибудь цветом, кроме перечисленных.**
Также я попрошу тебя **не менять, не передвигать и не удалять мои комментарии**, чтобы при последующей проверке мне было проще ориентироваться.
    
Кроме того, комментарии у меня обозначены как заголовки. Это сбивает содержание, но при последующей проверке я могу снять с них заголовок, если тебе так будет удобнее. При этом они удобны, поскольку ты сразу можешь перейти к ним.   
        
<hr>
    
<font color= 'dodgerblue'>**По проекту:**</font> работа выполнена очень хорошо, все оформлено понятно и аккуратно, есть анализ и промежуточные выводы. Видно, что ты уже уверенно пользуешься методами **pandas**, это здорово. Доли невозврата найдены верно. У меня остались некоторые вопросы, я все написала в комментариях. Там же я оставила и рекомендации по улучшению проекта. Буду ждать работу на повторное ревью.
    
<hr style="border-top: 3px solid purple; "></hr>
Комментарии можешь оставлять с помощью такого кода в Markdown ячейке:
    
    
    <div class="alert alert-info">
    <h2> Комментарий студента</h2>

    Тут текст.
    </div>
    
    
    <font color='red'> Этот код позволяет менять цвет текста. </font>     

<font color='red'> Выглядит это так. </font> 
  
    
Если не хочешь, чтобы твои комментарии отображались в содержании как заголовки, замени **h2** на **b** или добавь `<a class="tocSkip">` после фразы *Комментарий студента*.


О том, как **форматировать текст** в Markdown ячейке или как добавлять ссылки, ты можешь узнать [тут](https://sqlbak.com/blog/jupyter-notebook-markdown-cheatsheet) и [вот тут](https://medium.com/analytics-vidhya/the-ultimate-markdown-guide-for-jupyter-notebook-d5e5abf728fd).
</div>

<div style="border-radius: 15px; border: 3px solid indigo; padding: 15px;">
<b> Комментарий ревьюера 2</b>
    
Спасибо за корректировки. Я оставила новые комментарии с цифрой 2, а со старых сняла заголовки, чтобы  не мешали. Пара вопросов еще. 
</div>

<div style="border-radius: 15px; border: 3px solid indigo; padding: 15px;">
<b> Комментарий ревьюера 3</b>
    
Новые комментарии я уже не оставляла. 
    
    
Надеюсь, что  мои комментарии были или еще будут полезны.
    
Работу я могу принять, поскольку вопросов больше нет. Желаю успехов в дальнейшем обучении :)
</div>

# Исследование надежности заемщиков.

<div class="alert alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ❌</b>
    
Вводный параграф это неотъемлемая часть любого проекта, отчета, презентации или статьи. Будет очень здорово, если ты добавишь постановку задачи: краткое ее описание и описание данных. Это поможет не только тебе лучше ориентироваться в проекте спустя некоторое время, но и потенциальному работодателю, который будет просматривать твое портфолио. 
</div>

### Шаг 1. Обзор данных

<div class="alert alert-info">
<font color='black'> <b> В данном проекте нужно разобраться, влияет ли семейное положение, количество детей клиента, образование, доход клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов</font> <b> 
  



In [1]:
import pandas as pd
data=pd.read_csv('/datasets/data.csv')
data.info()
display(data.head(15))



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   

     total_income                            purpose  
0   253875.639453                      покупка жилья  
1   112080.014102            приобретение автомобиля  
2   145885.952297                      покупка жилья  
3   267628.550329         дополнительное образование  
4   158616.077870                    сыграть свадьбу  
5   255763.565419                      покупка жилья  
6   240525.971920                  операции с жильем  
7   135823.934197                        образование  
8    95856.832424              на проведение свадьбы  
9   144425.938277            покупка жилья для семьи  
10  113943.491460               покупка недвижимости  
11   77069.234271  покупка коммерческой недвижимости  
12            NaN                    сыграть свадьбу  
13  130458.228857            приобретение автомобиля  
14  165127.911772         покупка жилой недвижимости

<div class="alert alert-info">
<font color='black'> <b> По открытой таблице с данными можно сделать следующие выводы:  
   В данных имеетя 12 колонок, в двух колонках значений меньше, чем в остальных.
   Присутствуют целые числа, дробные числа и значения типа "object" </font> <b> 
   
    

<div class="alert alert-warning" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ⚠️</b>
    
Не забывай и про промежуточные выводы. 
    
</div>
<div class="alert alert-success" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера 2 ✔️</b>
    
Супер 👍

</div>

### Шаг 2.1 Заполнение пропусков

In [2]:
data.isna().sum()
median_days_employed=data['days_employed'].median()
median_total_income=data['total_income'].median()
data['days_employed']=data['days_employed'].fillna(median_days_employed)
data['total_income']=data['total_income'].fillna(median_total_income)
display(data.head(15))

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0   -1203.369529         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   

     total_income                            purpose  
0   253875.639453                      покупка жилья  
1   112080.014102            приобретение автомобиля  
2   145885.952297                      покупка жилья  
3   267628.550329         дополнительное образование  
4   158616.077870                    сыграть свадьбу  
5   255763.565419                      покупка жилья  
6   240525.971920                  операции с жильем  
7   135823.934197                        образование  
8    95856.832424              на проведение свадьбы  
9   144425.938277            покупка жилья для семьи  
10  113943.491460               покупка недвижимости  
11   77069.234271  покупка коммерческой недвижимости  
12  145017.937533                    сыграть свадьбу  
13  130458.228857            приобретение автомобиля  
14  165127.911772         покупка жилой недвижимости

<div class="alert alert-info">
<font color='black'> <b>Имеются пропуски типа Nan в двух столбцах 'days_employed' и 'total_income'. Пропуски составляют около 10% от количества значений по каждому из этих столбцов. Ошибки могли быть вызваны человеческим фактором, либо некорректное считывание, копирование. Заполнить пропуски медианными значениями нужно для того, чтобы минимизировать влияние пропущенных значений на общую картину данных*</font> <b> 

<div class="alert alert-success" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера ✔️</b>
    
Хорошо. Можно также заполнить медианой, но взятой по нескольким признакам. 

</div>


### Шаг 2.2 Проверка данных на аномалии и исправления.

In [3]:
data['days_employed'] = pd.to_numeric(data['days_employed']) 
data['total_income'] = pd.to_numeric(data['total_income'])
data['days_employed'] = data['days_employed'].abs()
data=data.loc[(data['dob_years']>10) & (data['gender']!='XNA')]
data.info()
data['children'] = data['children'].replace(-1, 1)
data['children'] = data['children'].replace(20, 2)

#display(data['children'].value_counts())
#display(data['dob_years'].value_counts())
#display(data['gender'].value_counts())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21423 entries, 0 to 21524
Data columns (total 12 columns):
children            21423 non-null int64
days_employed       21423 non-null float64
dob_years           21423 non-null int64
education           21423 non-null object
education_id        21423 non-null int64
family_status       21423 non-null object
family_status_id    21423 non-null int64
gender              21423 non-null object
income_type         21423 non-null object
debt                21423 non-null int64
total_income        21423 non-null float64
purpose             21423 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.1+ MB


<div class="alert alert-info">
<font color='black'> <b>В столбце 'days_employed' присутствовали отрицательные значения и значения, которых не может существовать (300000 рабочих дней существовать не может). В столбце 'children' есть отрицательное значение и аномально большое значение равное 20. Заменил отрицательное значение на такое же положительное, аномально большое значение заменил на среднее. Их не так много, большой роли не сыграет. Есть странное значение в столбце "пол" в количестве 1 шт., удалим (всего одно значение!). Возраста "0" не бывает, удалил эти значения (их сравнительно мало)</font> <b> 

<div class="alert alert-success" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ✔️</b>
    
- Соглашусь, по модулю числа в столбце стажа кажутся нормальными. Так что минус мог быть просто тире. А вот огромные вызывают много вопросов. Это могут быть часы, а могут быть и пропуски, которые при выгрузке трансформировались в аномальные для конкретного столбца числа.
    
    
- Ну да. Учитывая отсутствие клиентов с 6-19 детьми, а так же число 76, я склоняюсь к тому, что это ошибка. Скорее всего там либо 2, либо 20 это маркер пропущенного значения.
    
</div>
<div class="alert alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ❌</b>
    
- Есть ли еще какие-либо аномалии в данных? Например, в столбце возраста или пола.
<div class="alert alert-info">
    <h2> Комментарий студента</h2>

    Пропустил( Добавил в описание выше + добавил условия в код
    </div>


- Почему 20 заменяется на 3? Это ведь не среднее. 
<div class="alert alert-info">
    <h2> Комментарий студента</h2>

    Согласен. Имел ввиду что 3 это среднее количество между 1 и 5=) Среднее по количеству значений составило около 2600, медиана около 1000. Наверно лучше взять медиану, она окажется между количеством детей 2 и 3. Двое детей встречается чаще, так что исправим на 2
    </div>


- Строка `data=data.loc[data['days_employed']<10000]` удаляет очень много строк.  
<div class="alert alert-info">
    <h2> Комментарий студента</h2>
Почему? в году около 200 рабочих дней (ну плюс минус). 10000 дней это 50 лет! я и так много взял=) Изменил число на 300000, но все равно удаляется 3000 строк. Заменить аномальные значения > 10000 на медианные? напиши пожалуйста код как это сделать тогда, туплю
    </div>
    </div>

<div class="alert alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера 2 ❌</b>
    
    
Потому что ты удаляешь все строки, где стаж превышает указанное значение. А их слишком много, чтобы просто взять и удалить. Очевидно, что аномалия в стаже это какая-то ошибка выгрузки или склейки таблиц. Удаляя эти строки, мы лишаемся другой, не менее важной информации о клиентах. Если оставлять в таблице только строки с адекватным стажем, то придется удалять всю таблицу, т.к. адекватного стажа у нас в принципе нет. Так что удалять их не надо. 
    <div class="alert alert-info">
    <h2> Комментарий студента</h2>
Оставил строки
    </div>

    
    
    
    
</div>

### Шаг 2.3. Изменение типов данных.

In [4]:
data['total_income']=data['total_income'].astype('int')
data['days_employed']=data['days_employed'].astype('int')

display(data.head())

children  days_employed  dob_years education  education_id  \
0         1           8437         42    высшее             0   
1         1           4024         36   среднее             1   
2         0           5623         33   Среднее             1   
3         3           4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

### Шаг 2.4. Удаление дубликатов.

In [5]:
display(data.duplicated().sum())
data['education']=data['education'].str.lower()
data=data.drop_duplicates()





54

<div class="alert alert-info">
<font color='black'> <b>Был применен метод 'duplicated().sum()' для поиска количества строк-дубликатов
Было найдены строки с одинаковыми значениями в столбцах 'days_employed' и 'total_income', которые были удалены. В столбце 'education' значения приведены к единому регистру<b>

<div class="alert alert-warning" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ⚠️</b>
    
Имеет смысл сначала привести строки к нижнему регистру, а уже потом удалять дубликаты.</div>
<div class="alert alert-info">
    <h2> Комментарий студента</h2>
Исправил
    </div>
    </div>

### Шаг 2.5. Формирование дополнительных датафреймов словарей, декомпозиция исходного датафрейма.

In [6]:
education_data = data[['education_id','education']]
family_data = data[['family_status_id','family_status']]
#data = data[['children','days_employed','dob_years', 'education_id', 'family_status_id', 'gender', 'income_type', 'debt', 'total_income', 'purpose']]
education_data=education_data.drop_duplicates().reset_index(drop=True)
family_data=family_data.drop_duplicates().reset_index(drop=True)
data=data.drop('education', axis=1)
data=data.drop('family_status', axis=1)
display (education_data.head(10))
display (family_data.head(10))
display (data.head(10))


education_id            education
0             0               высшее
1             1              среднее
2             2  неоконченное высшее
3             3            начальное
4             4       ученая степень

family_status_id          family_status
0                 0        женат / замужем
1                 1       гражданский брак
2                 2         вдовец / вдова
3                 3              в разводе
4                 4  Не женат / не замужем

children  days_employed  dob_years  education_id  family_status_id gender  \
0         1           8437         42             0                 0      F   
1         1           4024         36             1                 0      F   
2         0           5623         33             1                 0      M   
3         3           4124         32             1                 0      M   
4         0         340266         53             1                 1      F   
5         0            926         27             0                 1      M   
6         0           2879         43             0                 0      F   
7         0            152         50             1                 0      M   
8         2           6929         35             0                 1      F   
9         0           2188         41             1                 0      M   

  income_type  debt  total_income                     purpose  
0   сотрудник     0        253875               покупка жилья  
1   сотрудник     0        112080     приобретение автомобиля  
2   сотрудник     0        145885               покупка жилья  
3   сотрудник     0        267628  дополнительное образование  
4   пенсионер     0        158616             сыграть свадьбу  
5   компаньон     0        255763               покупка жилья  
6   компаньон     0        240525           операции с жильем  
7   сотрудник     0        135823                 образование  
8   сотрудник     0         95856       на проведение свадьбы  
9   сотрудник     0        144425     покупка жилья для семьи

<div class="alert alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ❌</b>
    
- Как видно, в словарях есть дубликаты. Их нужно удалить. 

<div class="alert alert-info">
<h2> Комментарий студента</h2>
Удалил дубликаты</div>
    
    
    
    
- Нас также просят удалить столбцы из таблицы. 

<div class="alert alert-info">
<h2> Комментарий студента</h2>
Выполнено</div>
    
</div>    
<div class="alert alert-success" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера 2  ✔️</b>
    
Все верно.

</div>    

### Шаг 2.6. Категоризация дохода.

In [7]:
def category_income(total_income):
    if 0<=total_income<= 30000:
        return 'E'
    elif 30001<=total_income<= 50000:
        return 'D'
    elif 50001<=total_income<= 200000:
        return 'C'
    elif 200001<=total_income<= 1000000:
        return 'B'
    return 'A'
data['total_income_category']=data['total_income'].apply(category_income)
display (data['total_income_category'].value_counts()) 

def purpose_category(category):
    if 'автомоб' in category:
        return 'операции с автомобилем'
    elif 'свадь' in category:
        return 'проведение свадьбы'
    elif 'образова' in category:
        return 'получение образования'
    elif 'недвижим' or 'жиль' in category:
        return 'операции с недвижимостью'
    
data['purpose_category']=data['purpose'].apply(purpose_category)
display (data['purpose_category'].value_counts())
display (data.head(10))


C    15943
B     5014
D      348
A       25
E       22
Name: total_income_category, dtype: int64

операции с недвижимостью    10763
операции с автомобилем       4284
получение образования        3995
проведение свадьбы           2310
Name: purpose_category, dtype: int64

children  days_employed  dob_years  education_id  family_status_id gender  \
0         1           8437         42             0                 0      F   
1         1           4024         36             1                 0      F   
2         0           5623         33             1                 0      M   
3         3           4124         32             1                 0      M   
4         0         340266         53             1                 1      F   
5         0            926         27             0                 1      M   
6         0           2879         43             0                 0      F   
7         0            152         50             1                 0      M   
8         2           6929         35             0                 1      F   
9         0           2188         41             1                 0      M   

  income_type  debt  total_income                     purpose  \
0   сотрудник     0        253875               покупка жилья   
1   сотрудник     0        112080     приобретение автомобиля   
2   сотрудник     0        145885               покупка жилья   
3   сотрудник     0        267628  дополнительное образование   
4   пенсионер     0        158616             сыграть свадьбу   
5   компаньон     0        255763               покупка жилья   
6   компаньон     0        240525           операции с жильем   
7   сотрудник     0        135823                 образование   
8   сотрудник     0         95856       на проведение свадьбы   
9   сотрудник     0        144425     покупка жилья для семьи   

  total_income_category          purpose_category  
0                     B  операции с недвижимостью  
1                     C    операции с автомобилем  
2                     C  операции с недвижимостью  
3                     B     получение образования  
4                     C        проведение свадьбы  
5                     B  операции с недвижимостью  
6                     B  операции с недвижимостью  
7                     C     получение образования  
8                     C        проведение свадьбы  
9                     C  операции с недвижимостью

### Шаг 2.7. Категоризация целей кредита.

In [8]:
total_income_category_count=data.groupby('total_income_category')['debt'].count()
total_income_category_sum=data.groupby('total_income_category')['debt'].sum()
total_income_conversion=total_income_category_sum/total_income_category_count
display(total_income_conversion)
children_count=data.groupby('children')['debt'].count()
children_sum=data.groupby('children')['debt'].sum()
children_conversion=children_sum/children_count
display(children_conversion.sort_values( ascending=False))
education_count=data.groupby('education_id')['debt'].count()
education_sum=data.groupby('education_id')['debt'].sum()
education_conversion=education_sum/education_count
display (education_conversion.sort_values( ascending=False))
family_count=data.groupby('family_status_id')['debt'].count()
family_sum=data.groupby('family_status_id')['debt'].sum()
family_conversion=family_sum/family_count
display (family_conversion.sort_values( ascending=False))
purpose_count=data.groupby('purpose_category')['debt'].count()
purpose_sum=data.groupby('purpose_category')['debt'].sum()
purpose_conversion=purpose_sum/purpose_count
display (purpose_conversion.sort_values( ascending=False))




total_income_category
A    0.080000
B    0.070802
C    0.084865
D    0.060345
E    0.090909
Name: debt, dtype: float64

children
4    0.097561
2    0.095553
1    0.091341
3    0.082317
0    0.075458
5    0.000000
Name: debt, dtype: float64

education_id
3    0.109929
2    0.091768
1    0.089820
0    0.053116
4    0.000000
Name: debt, dtype: float64

family_status_id
4    0.097709
1    0.093485
0    0.075427
3    0.071730
2    0.064990
Name: debt, dtype: float64

purpose_category
операции с автомобилем      0.093371
получение образования       0.092616
проведение свадьбы          0.079654
операции с недвижимостью    0.072378
Name: debt, dtype: float64

<div class="alert alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера 2 ❌</b>
    
Не забывай перепроверять работоспособность ячеек перед отправкой проекта и после корректировок :)
<div class="alert alert-info">
    <h2> Комментарий студента</h2>
Исправил
    </div>

</div>

<div class="alert alert-success" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ✔️</b>
    
Обрати внимание на то, что ты делишь сумму на количество, а это не что иное, как среднее. Действия у тебя верные. В столбце `debt` единицами обозначено наличие задолженности. Нулем — ее отсутствие. Так, сумма столбца будет равна сумме единичек или числу должников, а применение метода **count** вернет общее количество кредитов. Таким образом, для поиска доли должников надо разделить сумму на количество. Можно это сделать в одной группировке (та же сводная):
</div>

In [ ]:
# КОД РЕВЬЮЕРА

data.groupby('children')['debt'].agg(['count','sum','mean'])

In [ ]:
# КОД РЕВЬЮЕРА

def my_mean(x): return x.mean()*100

# В кач-ве агрегирования можно любую функцию брать.
data.groupby('children')['debt'].agg(['count', 'sum', my_mean])

In [ ]:
# КОД РЕВЬЮЕРА

def my_mean(x): return '{:.2%} '.format(x.mean())

data.groupby('children')['debt'].agg(['count', 'sum', my_mean])

### Ответы на вопросы.

##### Вопрос 1:

<div class="alert alert-info">
<font color='black'> <b>1) Есть ли зависимость между количеством детей и возвратом кредита в срок?
У тех людей, у которых больше детей, больше вероятность просрочки по кредитам
2) Есть ли зависимость между семейным положением и возвратом кредита в срок?
У тех людей, кто потерял супруга или в разводе, меньше вероятность просрочки по кредитам
3) Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
Большой разницы судя по всему нет 
4) Как разные цели кредита влияют на его возврат в срок?
У тех, кто проводит операции с авто, больше вероятность просрочки.
У тех, кто проводит операции с недвижимостью, эта вероятность меньше всех<b>



##### Вывод 1:

<div class="alert alert-success" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<b> Комментарий ревьюера  ✔️</b>
    
    
- Дети и непредвиденные расходы это почти синонимы :)
    
    
- Овдовевшие это, как правило,  возрастные люди, а они более ответственные. Да и скорее всего у них уже есть взрослые дети, которые могут помогать с выплатами. В целом, можно сказать так: самые безответственные те, кто никогда не был в браке. 
</div>

## Общий вывод:

<div class="alert alert-info" >
<font color='black'> <b>
    Наибольшую вероятность просрочки имеют следующие люди:
    1) Имеющие 4 детей - 9.7 %
    2) Имеющие начальное образование - 10.9%
    3) Не состоящие в официальных отношениях - 9.7%
    4) Берущие кредит на операции с автомобилем - 9.3%.
    Наименьшую вероятность просрочки имеют следующие люди:
    1) Имеющие 0 детей - 7.5 %
    2) Имеющие высшее образование - 5.3%
    3) Вдовцы и вдовы - 6.4%
    4) Берущие кредит на операции с недвижимостью - 7.2%  <b>